kasa/cli.py

```diff
@@ -427,8 +427,8 @@ async def print_discovered(dev: SmartDevice):
                 echo()
             else:
                 discovered[dev.host] = dev.internal_state
-                ctx.obj = dev
-                await ctx.invoke(state)
+                ctx.parent.obj = dev
+                await ctx.parent.invoke(state)
```

Comment:
This feels unrelated to test coverage improvements.

Reply:
Two of the uncovered pieces of code were simply properties of the `TapoDevice`, i.e. `features` and `device_id`.  This change brought in some code to really use those so they could be put into a sensible test.